torch/csrc/distributed/rpc/testing/faulty_process_group_agent.cpp

```diff
@@ -14,37 +18,44 @@ FaultyProcessGroupAgent::FaultyProcessGroupAgent(
     int numSendRecvThreads,
     std::chrono::milliseconds rpcTimeout,
     const std::vector<std::string>& messagesToFail,
+    const std::unordered_map<std::string, float>& messageTypesToDelay,
     int failNumSends)
     : ProcessGroupAgent(
           std::move(workerName),
           std::move(pg),
           numSendRecvThreads,
           rpcTimeout),
       failNumSends_(failNumSends),
-      messageTypesToFail_(parseMessagesToFailInput(messagesToFail)) {}
+      messageTypesToFail_(parseMessagesToFailInput(messagesToFail)),
+      messageTypesToDelay_(parseMessagesToDelay(messageTypesToDelay)) {}
 
 std::vector<MessageType> FaultyProcessGroupAgent::parseMessagesToFailInput(
     const std::vector<std::string>& messagesToFail) const {
   // Since we can only pass strings corresponding to the Message Types from the
   // python tests, we must parse the list of strings and resolve the actual
   // types. We will then check this list of types in the send function to
   // determine whether we should fail or not.
-  std::vector<MessageType> messageTypesToFail;
+  std::vector<MessageType> messageTypesToFail(messagesToFail.size());
```

Comment:
Should this be 

```c++
std::vector<MessageType> messageTypesToFail;
messageTypesToFail.reserve(messagesToFail.size());
```

Otherwise, the `push_back` below will start from `messagesToFail.size() + 1`'s item?

Reply:
Good point, will update